aws/resource_aws_lb_listener_rule.go

```diff
@@ -920,3 +1234,115 @@ func highestListenerRulePriority(conn *elbv2.ELBV2, arn string) (priority int64,
 
 	return
 }
+
+func lbListenerRuleConditions(conditions []interface{}) ([]*elbv2.RuleCondition, error) {
+	elbConditions := make([]*elbv2.RuleCondition, len(conditions))
+	for i, condition := range conditions {
+		conditionMap := condition.(map[string]interface{})
+		conditionValues := conditionMap["values"].([]interface{})
+
+		elbConditions[i] = &elbv2.RuleCondition{
+			Field: aws.String(conditionMap["field"].(string)),
+		}
+		switch conditionMap["field"].(string) {
+		case "host-header":
+			hostHeader := conditionMap["host_header"].([]interface{})
+			var values []interface{}
+			if len(hostHeader) > 0 {
+				values = hostHeader[0].(map[string]interface{})["values"].([]interface{})
```

Comment:
Don't we know the type of it already?
```suggestion
				values = hostHeader[0].(map[string]string{})["values"].([]string{})
```

Reply:
Yes... and no. We might think we know what the type is but it's a data structure created by terraform. It's `interface{}`s all the way down. That slice is `[]interface{}` not `[]string`. I tried adding a debug statement with `%T`.